src/screens/login/useLogin.ts

```diff
@@ -0,0 +1,46 @@
+import {useRef} from 'react';
+import {Alert} from 'react-native';
+import {LoginScreenNavigationProp} from './types';
+import firestore from '@react-native-firebase/firestore';
+import '@react-native-firebase/app';
+
+export function useLogin({
+  navigation,
+}: {
+  navigation: LoginScreenNavigationProp;
+}) {
+  const usernameRef = useRef('');
+  const passwordRef = useRef('');
+  const usersCollection = firestore().collection('users');
+
+  async function isValidUser() {
+    try {
+      const userRef = usersCollection.doc(usernameRef.current);
+      const docSnapshot = await userRef.get();
+      if (docSnapshot.exists) {
+        const userData = docSnapshot.data();
+        if (userData?.password === passwordRef.current) {
+          navigation.navigate('FlatListToDoList');
+        } else {
+          Alert.alert('Invalid Credentials');
+        }
+      }
+    } catch (error) {
+      console.log(error);
```

Comment:
Besides the log, display a message to the user if something happened. The user needs to know that something happened.